### PR TITLE
Replace travis-ci by github actions for CI

### PR DESCRIPTION
**Pull request by prx0**
Hello!

I currently work to replace travis-ci by github actions. I build the project for stable, nightly and beta rust toolchain. Nightly and Beta jobs are allowed to fail because they used to prevent breaking changes in the code. 

This PR brings github actions for many stuff. I build and run tests in debug and release for rust stable, nightly and also beta to prevent breaking changes.
I add a coverage job which use tarpaulin to get the coverage value, codecov is used to make the badge. The workflow has jobs for cargo fmt, cargo clippy and to check the required minimal rustc version too.
I call cargo bench at the end of the process for rust stable pipeline to gets some benches.

It's ready for review if you want to migrate your project on github actions.

Thank you !

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # CircBuf
-
-[![Build Status](https://travis-ci.org/jeromefroe/circbuf-rs.svg?branch=master)](https://travis-ci.org/jeromefroe/circbuf-rs)
-[![codecov](https://codecov.io/gh/jeromefroe/circbuf-rs/branch/master/graph/badge.svg)](https://codecov.io/gh/jeromefroe/circbuf-rs)
+[![build status](https://github.com/prx0/circbuf-rs/actions/workflows/rust.yml/badge.svg)](https://github.com/prx0/circbuf-rs/actions/workflows/rust.yml)
```
Comment (jeromefroe) [inline README.md]: I think these URL's currently point to your fork and so we just need to replace `prx0` with `jeromefroe`.

**Review by jeromefroe — COMMENTED** [review]
This is amazing! Thank you for the PR @prx0! Had one small comment about the URL's of the badges but once that's fixed we should all set to merge.

File: README.md (hunk 1 above)
Comment (prx0) [reply README.md]: You're right! Thank you I changed these links.

**Comment by Stargateur** [comment]
Unless I miss it, you forget about `--all-features` to test against bytes feature. Use of `--all-features` is generally better than `--features bytes` cause it's very much advice that feature must be additive.

**Comment by Stargateur** [comment]
It's would be also nice to run benchmark and optionally have benchmark report, an action with `cargo +nightly bench --all-features -- --include-ignored` should do the job. `--include-ignored` include bench for target that have `is_read_vectored()` on `File` so any unix like.

https://github.com/benchmark-action/github-action-benchmark/blob/master/.github/workflows/rust.yml

**Comment by prx0** [comment]
> It's would be also nice to run benchmark and optionally have benchmark report, an action with `cargo +nightly bench --all-features -- --include-ignored` should do the job. `--include-ignored` include bench for target that have `is_read_vectored()` on `File` so any unix like.
> 
> https://github.com/benchmark-action/github-action-benchmark/blob/master/.github/workflows/rust.yml

Understood ! I add a job for cargo benchmark and I add the `--all-features` flag for tests.
cargo fmt and clippy are now in separate job and they not depends of build.

**Review by jeromefroe — APPROVED** [review]
Thank you @prx0!

**Pull request merged**